nautobot/dcim/forms.py

```diff
@@ -2566,13 +2588,16 @@ def __init__(self, *args, **kwargs):
         else:
             device = self.instance.device
 
-        # Limit LAG choices to interfaces belonging to this device or a peer VC member
-        device_query = Q(device=device)
-        if device.virtual_chassis:
-            device_query |= Q(device__virtual_chassis=device.virtual_chassis)
-        self.fields["lag"].queryset = Interface.objects.filter(
-            device_query, type=InterfaceTypeChoices.TYPE_LAG
-        ).exclude(pk=self.instance.pk)
+        # Restrict parent/bridge/LAG interface assignment by device
+        self.fields["parent_interface"].widget.add_query_param("device_id", device.pk)
+        self.fields["bridge"].widget.add_query_param("device_id", device.pk)
+        self.fields["lag"].widget.add_query_param("device_with_common_vc", device.pk)
```

Comment:
It's needed because the LAG filter was moved down to the field. And as mentioned above, it fails open and returns the device interfaces if not part of a VC. Candidate for refactor and fix in v2 IMO.